test/query.test.js

```diff
@@ -2447,11 +2447,20 @@ describe('Query', function() {
 
         let doc = yield Model.create({ n: 0 });
 
-        yield Model.findOneAndUpdate({ _id: doc._id }, { $inc: { n: 1 } }, err => {
+        let callbackDone;
```

Comment:
This seems a little convoluted. How about we just make it so that we call `Model.findOneAndUpdate()` with a callback and assert that the return value is a query? That seems like it will test what we need in a simpler way.

Reply:
@vkarpov15 Sure. I've submitted a new pull request which uses your way.

Reply:
I've submitted a new pull request at #6825